libcxx/test/std/ranges/range.adaptors/range.stride.view/concept.verify.cpp

```diff
@@ -0,0 +1,44 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20
+// expected-no-diagnostics
+
+// template <input_range _View> requires view<_View>
+
+#include "__ranges/stride_view.h"
+#include "almost_satisfies_types.h"
+#include "test_iterators.h"
+#include "test_range.h"
+#include "types.h"
+
+template <typename I, std::ranges::range_difference_t<I> D>
+concept CanStrideView = requires { std::ranges::stride_view<I>{I{}, D}; };
+
+// Ensure that the InputRangeNotIndirectlyReadable is a valid range.
+static_assert(std::ranges::range<InputRangeNotIndirectlyReadable>);
```

Comment:
These assertions are effectively testing the test. You should put them inside `types.h` instead, immediately after the place where you define `InputRangeNotIndirectlyReadable` & friends.